src/main/kotlin/com/cognifide/gradle/aem/instance/InstanceActions.kt

```diff
@@ -41,13 +44,23 @@ class InstanceActions(val project: Project) {
             progressLogger.progress(progressFor(instanceStates, config, timer))
 
             if (config.awaitTimes > 0 && timer.ticks > config.awaitTimes) {
-                logger.warn("Instance(s) are not stable. Timeout reached after ${Formats.duration(timer.elapsed)}")
+                logger.warn("Instance(s) are not stable. Timeout reached after ${Formats.duration(timer.elapsed)}.")
                 return@waitUntil false
             }
 
-            if (instanceStates.all { it.stable }) {
-                logger.info("Instance(s) are stable after ${Formats.duration(timer.elapsed)}")
-                return@waitUntil false
+            if (instanceStates.all(config.awaitCondition)) {
```

Comment:
The whole block is currently hard to understand. I opt for naming the conditions, or at least adding the comments

Reply:
this method has quite big complexity and even refactoring this in my opinion will not improve readability. 

yep, agree, I was thinking about naming conditions